conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ParametersHandler.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.palantir.conjure.java.undertow.lib.Parameters;
+import com.palantir.logsafe.Preconditions;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+
+/**
+ * {@link HttpHandler} that attaches to the exchange a safe and an unsafe parameters MultiMap such that a subsequent
+ * handler can attach parameters to the exchange using {@link Parameters} utility methods.
+ */
+public final class ParametersHandler implements HttpHandler {
+
+    private final HttpHandler delegate;
+
+    ParametersHandler(HttpHandler delegate) {
+        this.delegate = Preconditions.checkNotNull(delegate, "Delegate handler is required");
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
+        Multimap<String, Object> safeParams = HashMultimap.create();
+        Multimap<String, Object> unsafeParams = HashMultimap.create();
+        exchange.putAttachment(Parameters.SAFE_PARAMS_ATTACH_KEY, safeParams);
+        exchange.putAttachment(Parameters.UNSAFE_PARAMS_ATTACH_KEY, unsafeParams);
+        delegate.handleRequest(exchange);
```

Comment:
I don't think this handler is necessary, rather than logging a warning when the map is not present, the `Parameters` code can add a new map attachment. Not all endpoints have parameters, so we don't need to create or apply these maps to the exchange. Doing so is a bit dangerous because this handler could be applied twice, and lose references to previous maps.

Most services will have a mix of conjure-undertow and other handlers (like servlets) which will not apply these attachments, so our request logging implementation always needs to handle null attachment values.

Reply:
I'm fine with removing the handler